### PR TITLE
DE-3752 | sending edit events directly to Kinesis

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php [hunk 1]
```diff
@@ -375,10 +378,21 @@ public function setCategory() {
...
 
+	public function setBeacon( $beacon ) {
+		$this->mParams['beacon'] = $beacon;
+	}
+
 	public function sendLog() {
 		wfProfileIn( __METHOD__ );
 		$data = json_encode($this->mParams);
 		$this->getRabbit()->publish( $this->mKey, $data );
+		if ( ( ! Wikia::isDevEnv()) && ( $this->mParams['cityId'] % 100 == 28 ) ) { // enabled on 1% of wikis
```
Comment (macbre) [inline extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php]: Don't you want to use `<` operator instead and just increase the percentage of wikis as we go?

File: extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php [hunk 2]
```diff
@@ -375,10 +378,21 @@ public function setCategory() {
...
+			$this->mParams['action'] = $this->mKey;
+			$kinesis = new AsyncKinesisProducerTask();
+			( new AsyncTaskList() )
```
Comment (macbre) [inline extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php]: use `newLocalTask` static constructor instead. AFAIR you need to specify wiki ID when queuing a task. This constructor will use the ID of the current wiki.

File: extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php (hunk 1 above)
Comment (ghost) [reply extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php]: Good idea, I will use `<` indeed.

File: extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php (hunk 2 above)
Comment (ghost) [reply extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php]: OK, thanks!

**Review by jcellary — APPROVED** [review]

**Review by macbre — APPROVED** [review]

**Pull request merged**